### PR TITLE
Migrate es6-shim whatwg-fetch react-day-picker typings

### PR DESCRIPTION
**Pull request by leebyp**
- use typescript typings instead of `es6-shim`
- remove `whatwg-fetch` typings
- move `react-day-picker` typings to root `package.json` and use `@types` version

### CODE REVIEW

**Comment by adidahiya** [comment]
Instead of using different es6-shim types, can you try to use TypeScript's built-in `lib` types instead? I'm thinking of something like this in tsconfig.json:

```
    "lib": [
      "dom",
      "es2015.core",
      "es2015.collection",
      "es2015.generator",
      "es2015.iterable",
      "es2015.promise",
      "es2015.reflect",
      "es2015.symbol",
      "es5"
    ],
```

that would just be for table, mind you, which actually depends on es6-shim to polyfill those things.

**Comment by adidahiya** [comment]
also, what are we even using fetch for?

**Comment by giladgray** [comment]
whoa we actually don't use fetch here at all! it is used on internal docs for downloading some icons. should be safe to remove from this repo.

**Comment by adidahiya** [comment]
wait, that sounds like it _is_ used. `@types/fetch` is a dependency in packages/docs

**Comment by giladgray** [comment]
it's only used on the _internal_ docs site, [definitely not used here](https://github.com/palantir/blueprint/search?utf8=%E2%9C%93&q=fetch).

**Comment by adidahiya** [comment]
ah, gotcha, thanks for clarifying

**Comment by leebyp** [comment]
@adidahiya re-using the `--lib` flag are there any disadvantages to with using the full `es2015` (as in new commit)?

File: packages/table/src/common/grid.ts [hunk 1]
```diff
@@ -5,6 +5,8 @@
  * and https://github.com/palantir/blueprint/blob/master/PATENTS
  */
 
+import * as React from "react";
```
Comment (leebyp) [inline packages/table/src/common/grid.ts]: needs this now, because one of the methods here export a react interface through `Object.assign`...

**Comment by adidahiya** [comment]
@leebyp I haven't checked if all those types are polyfilled by es6-shim; it's _probably_ safe but one ought to double-check

File: packages/datetime/tsconfig.json [hunk 2]
```diff
@@ -19,7 +19,6 @@
         "target": "es5"
     },
     "include": [
-        "typings/tsd.d.ts",
```
Comment (giladgray) [inline packages/datetime/tsconfig.json]: check other tsconfigs for `typings/` paths and remove them too, if you can!

File: packages/table/tsconfig.json [hunk 3]
```diff
@@ -3,6 +3,17 @@
...
     "jsx": "react",
+    "lib": [
+      "dom",
+      "es2015.core",
+      "es2015.collection",
+      "es2015.generator",
+      "es2015.iterable",
+      "es2015.promise",
+      "es2015.reflect",
+      "es2015.symbol",
```
Comment (giladgray) [inline packages/table/tsconfig.json]: why are these needed here? does the docs use es6 features? or simply because it imports table?

**Comment by leebyp** [comment]
You're right, looks like `Proxy` isn't provided from `es6-shim`. This kinda sucks, we're in the middle of not enough compatibility or @types/es6-shim being out of date.

Side note: the `npm run build:gulp` command in circle has errors from unused variables. [This](https://github.com/Microsoft/TypeScript/issues/11734) fixes it but we'll need to be on TS 2.1 (and should also turn on `noUnusedLocals` in the remaining repos) - #492

**Comment by leebyp** [comment]
@giladgray i can't find the `fetch` dependency you guys are seeing?

File: packages/datetime/tsconfig.json (hunk 2 above)
Comment (leebyp) [inline packages/datetime/tsconfig.json]: yeah i checked the other ones, everything we can remove has been

File: packages/table/tsconfig.json (hunk 3 above)
Comment (leebyp) [inline packages/table/tsconfig.json]: yup, it's also the reason we get some errors in the build from unused variables (see below)

**Comment by giladgray** [comment]
@leebyp huh seems like `fetch` is only typings, not an actual dependency: https://github.com/palantir/blueprint/search?utf8=%E2%9C%93&q=fetch. so we're good 👍
